tests/test_segment_producer.py

```diff
@@ -0,0 +1,197 @@
+import collections
+import logging
+import os
+import pathlib
+import pickle
+import pytest
+import typing
+
+import intervaltree as itree
+
+import gdc_client.parcel.segment as segment
+import gdc_client.parcel.download_stream as stream
+import gdc_client.parcel.utils as utils
+
+directories_tuple = collections.namedtuple(
+    "directories", ["base_directory", "data_directory", "state_directory"]
+)
+
+data_info = collections.namedtuple("data_info", ["data", "md5sum"])
+
+
+@pytest.fixture()
+def setup_directories(tmp_path: pathlib.Path):
```

Comment:
Instead of setting up a bunch of temp files and such make this into an actual unit test and use mocks for the external calls. It will run faster and be easier to arrange things for testing scenarios.

Reply:
the very nature of segment producer loading state is to setup certain files and check for the existence of certain files. I think Mocking/patching would get really complex here: it might be faster, but I don't think speed is really an issue here